File: gcc/testsuite/gcc.dg/gomp/appendix-a/a.21.3.c
/* { dg-do compile } */

void work (int);

void
a21_good (int n)
{
  int i;
#pragma omp for ordered
  for (i = 0; i < n; i++)
    {
      if (i <= 10)
	{
#pragma omp ordered
	  work (i);
	}
      if (i > 10)
	{
#pragma omp ordered
	  work (i + 1);
	}
    }
}

File: gcc/testsuite/gfortran.dg/gomp/appendix-a/a.21.3.f90
! { dg-do compile }

      SUBROUTINE A21_GOOD(N)
      INTEGER N
!$OMP DO ORDERED
        DO I = 1,N
          IF (I <= 10) THEN
!$OMP ORDERED
              CALL WORK(I)
!$OMP END ORDERED
          ENDIF
          IF (I > 10) THEN
!$OMP ORDERED
              CALL WORK(I+1)
!$OMP END ORDERED
          ENDIF
        ENDDO
      END SUBROUTINE A21_GOOD
